cli/src/update.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{
+    config::{Config, ReleaseInfo},
+    error::WKCliError,
+    output::colored_print,
+    utils::compare_with_current_time,
+};
+use aion::*;
+use chrono::Utc;
+use clap::crate_version;
+use log::debug;
+use owo_colors::OwoColorize;
+use reqwest::Client;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+
+const WUKONG_GITHUB_REPO: &str = "mindvalley/wukong-cli";
+const GITHUB_API_URL: &str = "https://api.github.com";
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GithubLatestReleaseInfo {
+    pub tag_name: String,
+    pub url: String,
+    pub published_at: String,
+    pub html_url: String,
+}
+
+// check_for_update checks whether this wukong has had a newer release on Github
+pub async fn check_for_update() -> Result<(), WKCliError> {
+    let release_info = get_current_release_info()?;
+
+    if let Some(release_info) = release_info {
+        let last_update_checked_since =
+            compare_with_current_time(&release_info.checked_for_update_at);
+
+        if last_update_checked_since >= -(24.hours()) {
+            debug!("No need to check for update");
+            return Ok(());
+        }
+    }
+
+    debug!("Checking for update");
+
+    if let Some(latest_release_info) = get_latest_release_info(Some(GITHUB_API_URL)).await? {
+        let current_version = crate_version!().to_string();
+        let has_update = version_greater_than(&latest_release_info.version, &current_version);
+
+        if has_update {
+            debug!("New release found");
+            colored_print!(
+                "{} {} {} {}\n",
+                "A new release of wukong is available:".yellow(),
+                latest_release_info.version.cyan(),
+                "→".cyan(),
+                current_version.cyan()
+            );
+
+            colored_print!("To upgrade, run: brew upgrade wukong\n");
+            colored_print!("{}\n", latest_release_info.url.yellow());
+        } else {
+            debug!("No new release found");
+        }
+
+        update_release_info(latest_release_info)?;
+    }
+
+    Ok(())
+}
+
+fn get_current_release_info() -> Result<Option<ReleaseInfo>, WKCliError> {
```

Comment:
Not really sure if I like this or not, if we are not returning any error in this function, then I guess we don't need `Result` type, just `Option<ReleaseInfo>` will do.